omod/src/test/java/org/openmrs/module/oauth2login/web/controller/ProviderAttributeUserInfoTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.oauth2login.web.controller;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.openmrs.Provider;
+import org.openmrs.User;
+import org.openmrs.api.context.Context;
+import org.openmrs.util.PrivilegeConstants;
+
+public class ProviderAttributeUserInfoTest extends OAuth2IntegrationTest {
+	
+	@Override
+	protected String getAppDataDirName() {
+		// represents any IdP that allows to pass on the provider attribute as part of the user info JSON
+		return "anyidp";
+	}
+	
+	@Override
+	protected String getUserInfoJson() {
+		return "{\n" + "  \"sub\": \"4e3074d6-5e9f-4707-84f1-ccb2aa2ab3bc\",\n" + "  \"email_verified\": true,\n"
+		        + "  \"name\": \"Tommy Atkins\",\n" + "  \"preferred_username\": \"tatkins\",\n"
+		        + "  \"given_name\": \"Tommy\",\n" + "  \"family_name\": \"Atkins\",\n"
+		        + "  \"email\": \"tatkins@example.com\",\n" + "  \"roles\": [\"Provider\", \"B0gùs Rol3 N@m3\"]\n"
+		        + ", \"provider\": \"false\"}";
+	}
+	
+	@Override
+	protected void assertAuthenticatedUser(User user) {
+		Assert.assertEquals("tatkins", user.getUsername());
+		Assert.assertEquals("4e3074d6-5e9f-4707-84f1-ccb2aa2ab3bc", user.getSystemId());
+		Assert.assertEquals("Tommy", user.getGivenName());
+		Assert.assertEquals("Atkins", user.getFamilyName());
+		Assert.assertEquals("tatkins@example.com", user.getEmail());
+		assertProviderAccountDeactivation(user);
+	}
+	
+	@Test
+	public void assertOAuth2AuthenticationAndExistingProviderAccountDeactivationGivenProviderSetToFalse() throws Exception {
```

Comment:
This should become a standard test method naming. Maybe: `authenticationWithProviderSetAsFalse_shouldDeactivateProviderAccount`.